src/main/java/org/jenkinsci/plugins/workflow/graphanalysis/FlowScanningUtils.java

```diff
@@ -59,6 +62,53 @@ public boolean apply(FlowNode input) {
     // Default predicates, which may be used for common conditions
     public static final Predicate<FlowNode> MATCH_BLOCK_START = (Predicate)Predicates.instanceOf(BlockStartNode.class);
 
+    /** Sorts flownodes putting the one begun last (oldest startTime) at the end, with null times last
+     *  because likely they represent the newest nodes with a {@link TimingAction} not attached yet. */
```

Comment:
Yes, it is generally more useful and correct than sorting by ID (even though people insist on doing that anyway).  It is also intended for use in the 3rd level set of APIs that provide a tree-structured format with navigation -- I may have had to abandon that work, but don't see the point in ripping out the independently useful portions.